src/inet/InetInterface.cpp

```diff
@@ -255,7 +255,21 @@ bool InterfaceIterator::HasBroadcastAddress()
 
 CHIP_ERROR InterfaceIterator::GetInterfaceType(InterfaceType & type)
 {
-    return CHIP_ERROR_NOT_IMPLEMENTED;
+    VerifyOrReturnError(HasCurrent(), CHIP_ERROR_INCORRECT_STATE);
+
+    if(!mCurNetif)
+        return CHIP_ERROR_INCORRECT_STATE;
+
+    if (mCurNetif->flags & NETIF_FLAG_ETHARP)
```

Comment:
You can use `netif_is_flag_set` and it should be `NETIF_FLAG_ETHERNET`

Reply:
I've updated this, but it is also not working with NETIF_FLAG_ETHERNET